File: Data Structures & Algorithms/longest-common-prefix/submission-2.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        prefix = strs[0]

        for s in strs:
            i = 0
            while i < min(len(prefix), len(s)):
                if (prefix[i] != s[i]):
                    break
                i += 1
            prefix = prefix[0:i]

        return prefix

File: Data Structures & Algorithms/string-encode-and-decode/submission-2.java
class Solution {

    public String encode(List<String> strs) {
        StringBuilder encoded = new StringBuilder();
        for (String s : strs) {
            encoded.append(s.length()).append('#').append(s);
        }
        return encoded.toString();
    }

    public List<String> decode(String str) {
        List<String> decoded = new ArrayList<>();

        int i = 0;
        while (i < str.length()) {
            int j = str.indexOf('#', i);
            int length = Integer.parseInt(str.substring(i, j));
            j++; // move pass '#'
            decoded.add(str.substring(j, j + length));
            i = j + length;
        }

        return decoded;
    }
}
